docs/template_plugin/tests/functional/op_reference/grid_sample.cpp

```diff
@@ -0,0 +1,475 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/grid_sample.hpp"
+
+#include "base_reference_test.hpp"
+#include "gtest/gtest.h"
+
+using namespace ov;
+using namespace reference_tests;
+
+namespace {
+struct GridSampleParams {
+    GridSampleParams(const reference_tests::Tensor& data,
+                     const reference_tests::Tensor& grid,
+                     const op::v9::GridSample::Attributes& attrs,
+                     const reference_tests::Tensor& expected,
+                     const std::string& name)
+        : data_tensor{data},
+          grid_tensor{grid},
+          attributes{attrs},
+          expected_tensor{expected},
+          test_case_name{name} {}
+
+    reference_tests::Tensor data_tensor;
+    reference_tests::Tensor grid_tensor;
+    op::v9::GridSample::Attributes attributes;
+    reference_tests::Tensor expected_tensor;
+    std::string test_case_name;
+};
+
+class ReferenceGridSample : public testing::TestWithParam<GridSampleParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        const auto& params = GetParam();
+        function = CreateFunction(params.data_tensor, params.grid_tensor, params.attributes);
+        inputData = {params.data_tensor.data, params.grid_tensor.data};
+        refOutData = {params.expected_tensor.data};
+    }
+
+    static std::string getTestCaseName(const testing::TestParamInfo<GridSampleParams>& obj) {
+        return obj.param.test_case_name;
+    }
+
+private:
+    static std::shared_ptr<Model> CreateFunction(const reference_tests::Tensor& data,
+                                                 const reference_tests::Tensor& grid,
+                                                 const op::v9::GridSample::Attributes& attributes) {
+        const auto in1 = std::make_shared<op::v0::Parameter>(data.type, data.shape);
+        const auto in2 = std::make_shared<op::v0::Parameter>(grid.type, grid.shape);
+        const auto grid_sample = std::make_shared<op::v9::GridSample>(in1, in2, attributes);
+        return std::make_shared<Model>(NodeVector{grid_sample}, ParameterVector{in1, in2});
+    }
+};
+
+constexpr auto GS_BICUBIC{op::v9::GridSample::InterpolationMode::BICUBIC};
+constexpr auto GS_BILINEAR{op::v9::GridSample::InterpolationMode::BILINEAR};
+constexpr auto GS_NEAREST{op::v9::GridSample::InterpolationMode::NEAREST};
+
+constexpr auto GS_BORDER{op::v9::GridSample::PaddingMode::BORDER};
+constexpr auto GS_REFLECTION{op::v9::GridSample::PaddingMode::REFLECTION};
+constexpr auto GS_ZEROS{op::v9::GridSample::PaddingMode::ZEROS};
+
+constexpr std::array<op::v9::GridSample::PaddingMode, 3> padding_modes{GS_ZEROS, GS_BORDER, GS_REFLECTION};
+constexpr std::array<bool, 2> align_modes{false, true};
```

Comment:
LGTM! I would prefer to stick with matching variable and attribute names (`align_corners`)